### DISCUSSION TITLE
Policy for log levels? Give each non-DEBUG output an identifier.

### DISCUSSION THREAD

**Discussion by lab-at-nohl** [tine20/tine20#7430] [Ideas]
I recently started to use DBLogger, not in DEBUG of course. However, I haven't found a useful way to monitor my installation. It made me wondering, what the policy for log levels LIKE NOTICE, WARN, ERR and CRIT are...

Some examples of what I have as much in my logs that **logging is rendered useless**: 

- NOTICE: `Tinebase_Mail::decodingErrorHandler::611 stream_get_contents(): iconv stream filter (&quot;utf-8&quot;=&gt;&quot;utf-8&quot;): invalid multibyte sequence in /tine/vendor/zendframework/zendframework1/library/Zend/Mime/Part.php::346 (2)` (max. DEBUG, seems developer related)
- WARN: `Felamimail_Controller_Message::_updateFolderCounts::1337 folder counters dont match => refresh counters` (expected: NOTICE)
- WARN: `Syncroton_Command_Sync::handle::249 invalid synckey 4744 provided` (expected: NOTICE max. better DEBUG)
- CRIT: `Syncroton_Server::_handlePost::110 command not supported: ResolveRecipients` (expected: NOTICE max. better DEBUG)
- CRIT: `Syncroton_Server::_handlePost::166 exception message: SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction, query was: INSERT INTO tine20_acsync_content (device_id, folder_id, contentid, creation_time, creation_synckey, id) VALUES (?, ?, ?, ?, ?, ?)` (expected: DEBUG)

It seems that _everything stemming from Syncroton_Server_ class is CRIT automatically. 

On the other hand, **failed login attempts are only NOTICE** but I would expect them to be WARN at least?!

**Idea:** It would be helpful to give each Log Message an unique identifier. The logging facilities could have (beside the filter) a list of "always" and "never" to log ids.

**Comment by pschuele** [comment]
hi Johannes,
you are right - the log level usage is often quite random :(
I'll update the mentioned lines in tine.
it's not quite as easy in Syncroton: but I'll check that as well. the handPost logs might be CRITs after all - depending on the encountered problem? 

> On the other hand, failed login attempts are only NOTICE but I would expect them to be WARN at least?!

hm, this might easily spam the log file - I think NOTICE is ok here.

> Idea: It would be helpful to give each Log Message an unique identifier. The logging facilities could have (beside the filter) a list of "always" and "never" to log ids.

bot sure if this helps - I think it would be better to improve the code and use the correct log levels.
maybe this is a good foundation for what each level means: https://en.wikipedia.org/wiki/Syslog#Severity_level

here's some additional input on this topic:  https://sematext.com/blog/logging-levels/

**Comment by pschuele** [comment]
syncroton PR: https://github.com/tine20/syncroton/pull/30

**Comment by lab-at-nohl** [comment]
Thanks for the patches, which I've implemented manually in my current production installation. 

What do you think about `Felamimail_Controller_Message::_updateFolderCounts::1337 folder counters dont match => refresh counters` Warnings? Do you see any (severe) consequences in such cases that legitimate **WARN**? In my view it only consumes some CPU to get the values from IMAP again. Besides I don't see why in any case it is necessary to test for NULL values (`isset || array_key_exists`). After re-evaluating the function I would vote for DEBUG level here...

**Comment by pschuele** [comment]
yup, already fixed that: https://github.com/tine20/tine20/blob/main/tine20/Felamimail/Controller/Message.php#L1340 (https://github.com/tine20/tine20/commit/0c131775686931c4f4e16da922a3ddaf54e751bb).

sorry for the slow updates - the automatic merge was broken and I just fixed it :wink:
